plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+                op_modifies=True,
+                params={
+                    "virtual_network_name": vn_name,
+                },
+            )
+            response = response.get("response")
+            self.log("Received API response from 'get_layer3_virtual_networks' for vn '{0}': {1}".format(vn_name, str(response)), "DEBUG")
+
+            if not response:
+                self.log("Given layer3 Virtual Network '{0}' is not present in Cisco Catalyst Center.".format(vn_name), "INFO")
+                return False
+
+        except Exception as e:
+            self.msg = (
+                "Error while getting the details for layer3 virtual network '{0}' present in "
+                "Cisco Catalyst Center: {1}"
+            ).format(vn_name, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
+
+        return True
+
+    def get_fabric_ids(self, fabric_locations):
+        """
+        Retrieves a list of fabric IDs based on the specified fabric locations.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            fabric_locations (list): A list of dictionaries, each containing information about fabric locations.
+                including "site_name_hierarchy" and "fabric_type". The "fabric_type" can either be "fabric_site" or
+                "fabric_zone".
+        Returns:
+            list: A list of fabric IDs corresponding to the specified fabric locations. If a fabric ID cannot
+                be retrieved for a location, a warning is logged, and the function continues to the next location.
+        Description:
+            This function processes a list of fabric locations to extract fabric IDs. For each location, it
+            validates the fabric type and retrieves the corresponding site ID. Depending on the fabric type, it
+            calls either `get_fabric_site_id` or `get_fabric_zone_id` to obtain the fabric ID.
+            If the site is not recognized as a fabric site or zone, a warning message is logged, and that
+            location is skipped. The resulting list of fabric IDs is returned.
+        """
+
+        fabric_id_list = []
+
+        for fabric in fabric_locations:
+            site_name = fabric.get("site_name_hierarchy")
+            fabric_type = fabric.get("fabric_type", "fabric_site")
+            # Validate the correct fabric_type given in the playbook
+            self.validate_fabric_type(fabric_type).check_return_status()
+            site_id = self.get_site_id(site_name)
+
+            if fabric_type == "fabric_site":
+                fabric_id = self.get_fabric_site_id(site_name, site_id)
+            else:
+                fabric_id = self.get_fabric_zone_id(site_name, site_id)
+
+            if not fabric_id:
+                self.log("Unable to get the Fabric Id as the give site '{0}' is not fabric site/zone".format(site_name), "WARNING")
```

Comment:
"Unable to retrieve the Fabric ID for site '{0}' as it is not a recognized fabric site/zone.".format(site_name),